homeassistant/components/hassio/ingress.py

```diff
@@ -134,7 +134,7 @@ async def _handle_request(
             if (
                 hdrs.CONTENT_LENGTH in result.headers
                 and int(result.headers.get(hdrs.CONTENT_LENGTH, 0)) < 4194000
-            ):
+            ) or result.status in (204, 304):
```

Comment:
This should use `(HTTPStatus.NO_CONTENT, HTTPStatus.NOT_MODIFIED)` and should have a test that proves that these 2 statuses return `Response` and not `StreamResponse`